presto-blackhole/src/main/java/com/facebook/presto/plugin/blackhole/BlackHoleMetadata.java

```diff
@@ -161,7 +161,31 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
     @Override
     public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
     {
-        return new BlackHoleOutputTableHandle(new BlackHoleTableHandle(tableMetadata));
+        int splitsCount = (Integer) tableMetadata.getProperties().get(SPLITS_COUNT_PROPERTY);
+        int pagesPerSplit = (Integer) tableMetadata.getProperties().get(PAGES_PER_SPLIT_PROPERTY);
+        int rowsPerPage = (Integer) tableMetadata.getProperties().get(ROWS_PER_PAGE_PROPERTY);
+
+        if (splitsCount > 0 || pagesPerSplit > 0 || rowsPerPage > 0) {
```

Comment:
This doesn't handle negative values. It might be better to write it as:

```
checkArgument((splitsCount > && pagesPerSplit > 0 && rowsPerPage > 0) || (splitsCount == 0 && pagesPerSplit == 0 || rowsPerPage == 0), ...)
```

I think it also makes the intent clearer.